controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -482,16 +482,19 @@ private DeployOptions withVersion(Version version, DeployOptions options) {
     /** Register a DNS name for rotation */
     private void registerRotationInDns(Rotation rotation, String dnsName) {
         try {
-            Optional<Record> record = nameService.findRecord(Record.Type.CNAME, RecordName.from(dnsName));
+
             RecordData rotationName = RecordData.fqdn(rotation.name());
-            if (record.isPresent()) {
+            List<Record> records = nameService.findRecords(Record.Type.CNAME, RecordName.from(dnsName));
+            records.forEach(record -> {
```

Comment:
This isn't really possible. If it does happen it's an error in the `NameService` implementation, but that should be detected and thrown here. There can be only one CNAME record for a given name.

Reply:
That is exactly what happens here. In Route53 we expect two cnames for a given name - you can have as many as you like, but I don't see any purpose for more than two (one for the public zone file and one private zone file).

Reply:
What do mean by "two cnames"? One from each of Route53 and DnsDB?

Reply:
Two from route53. 

I believe generally name services can operate with multiple zone files. Only one is active unless you have a split view where you would have 2 zone files active. This (the split view) is the case in Route53. 

So unless you specify which zone file you are querying you get potentially one cname from each zone.

Reply:
Ok, that was not clear as the current interface doesn't really allow this. `findRecords(zone, type, name)` would be a better interface.

I guess this is fine for now though.